src/pages/RecipeCalculator.vue

```diff
@@ -79,10 +104,31 @@ export default defineComponent({
     GameAsset,
     RecipeTree
   },
+  methods: {
+    listMaterials(tree:RecipeObject) {
+      // Setup result
+      let result: Record<string, number> = {};
+      function flatten(current:RecipeObject) {
+        if(current.materials) {
+          current.materials.forEach(mat => {
+            flatten(mat);
+          })
+        } else {
+          let name = current.name;
+          let quantity = current.quantity;
+          if(result.hasOwnProperty(name)) { result[name] += quantity }
+          else { result[name] = quantity }
+        }
+      }
+      flatten(tree);
+      return result;
+    }
+  },
   setup() {
     const data: Record<string, RecipeObject> = calculatorData;
     const recipe = ref("");
     const quantity = ref("");
```

Comment:
This shouldn't be a string. I would suggest a default value of 1 (`ref(1)`), because most people probably want that as a default.

Reply:
Not sure what the change is mean to be

Reply:
```suggestion
    const quantity = ref(1);
```

Reply:
This means you also don't have to cast `Number(quantity)` in the template, just `quantity`.

Reply:
Quantity updated to `ref(1)` and Display updated to `ref("list")`